src/app/mail/mail-sidebar/compose-mail/compose-mail.component.ts

```diff
@@ -1096,6 +1089,27 @@ export class ComposeMailComponent implements OnInit, AfterViewInit, OnChanges, O
     }
   }
 
+  addHyperLink() {
+    var regex = /(https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|www\.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9]+\.[^\s]{2,}|[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.com|www\.[a-zA-Z0-9]+\.[^\s]{2,})/gi;
+    this.quill.focus();
+    var contents = this.quill.getText();
+    var filtered = contents.trim().split(/\s+/);
+    for (let i = 0; i < filtered.length; i++) {
+      var match = filtered[i].match(regex);
+      if (match !== null) {
+        var url = match[0];
+        var hyperLink = url;
+        if (!/^https?:\/\//i.test(url)) {
+          hyperLink = `http://${url}`;
```

Comment:
Are we replacing HTTPS with HTTP?

Reply:
NO. We add http to non-http links.
For example, google.com -> http://google.com